frontend/tet/admin/src/store/PreviewContext.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from 'react';
+import { TetData } from 'tet-shared/types/TetData';
+import TetPosting from 'tet/admin/types/tetposting';
+
+type PreviewContextObj = {
+  showPreview: boolean;
+  tetPosting: TetPosting;
+  setPreviewVisibility: (visibility: boolean) => void;
+  setTetPostingData: (data: TetPosting) => void;
+  getTemplateData: () => TetData;
+};
+
+export const initialPosting: TetPosting = {
+  title: '',
+  description: '',
+  location: {
+    name: '',
+    label: '',
+    value: '',
+    street_address: '',
+    postal_code: '',
+    city: '',
+  },
+  org_name: '',
+  spots: 1,
+  start_date: '',
+  contact_email: '',
+  contact_first_name: '',
+  contact_last_name: '',
+  date_published: null,
+  contact_language: 'fi',
+  contact_phone: '',
+  keywords: [],
+  keywords_working_methods: [],
+  keywords_attributes: [],
+};
+const initialTemplateData: TetData = {
+  title: '',
+  description: '',
+  location: {
+    name: '',
+    street_address: '',
+    postal_code: '',
+    city: '',
+  },
+  org_name: '',
+  spots: 1,
+  start_date: '',
+  contact_email: '',
+  contact_first_name: '',
+  contact_last_name: '',
+  date_published: null,
+  contact_language: '',
+  contact_phone: '',
+  keywords: [],
+  keywords_working_methods: [],
+  keywords_attributes: [],
+};
+
+export const PreviewContext = React.createContext<PreviewContextObj>({
```

Comment:
Perhaps I'm not just big fan of React.context so I'm thinking do we absolutely need this context? What happens if thepage is refreshed? Does the data just get lost or is it retained from backend + react-query? If it's derived from react-query data, then you might not need it but you could use select function to derive value from useQuery.
Some ideas: https://tkdodo.eu/blog/react-query-as-a-state-manager
I have actually used just query params or `useLocalstorage` to save user settings or that part of application state which is not synced to backend. If losign data on page refresh is not a problem then useContext is probably just fine :)